shared/middleware/helpers/storage_apps.rb

```diff
@@ -204,6 +214,27 @@ def reset_abuse(channel_id)
     0
   end
 
+  #
+  # Disables or enables automated content moderation for this project by
+  # altering the value for skip_content_moderation.
+  # @param [String] channel_id - an encrypted channel id
+  # @param [Boolean] should_skip, whether the content moderation should be
+  # skipped or not for this project.
+  # @raise [NotFound] if the channel does not exist or already has the desired
+  # value for skip_content_moderation
+  #
+  def change_content_moderation(channel_id, should_skip)
+    _owner, id = storage_decrypt_channel_id(channel_id)
+
+    row = @table.where(id: id).exclude(state: 'deleted').first
+    raise NotFound, "channel `#{channel_id}` not found" unless row
+
+    update_content_moderation = @table.where(id: id).exclude(state: 'deleted').update({skip_content_moderation: should_skip})
+    raise NotFound, "channel `#{channel_id}` not found" if update_content_moderation == should_skip
```

Comment:
I don't think this works the way you describe it (there's no test for it) and I'm not sure we want to do it anyway.

[Sequel::Dataset#update](https://sequel.jeremyevans.net/rdoc/classes/Sequel/Dataset.html#method-i-update) returns the number of rows updated, not the new value; therefore `update_content_moderation == should_skip` is probably never true becase it's comparing an integer to a boolean.

It's generally a good idea to make server APIs idempotent (I enjoyed [this article](https://stripe.com/blog/idempotency) on the subject).  In this case, that means we want this method to do nothing and report success if the setting is already the desired value, rather than throw an error.  This way, if a client requests this operation but the server's response gets lost on the way back, the client can safely re-send the request and not worry about causing an error.